pandas/tests/io/test_excel.py

```diff
@@ -136,23 +151,28 @@ def test_usecols_int(self, ext):
         # usecols as int
         with tm.assert_produces_warning(FutureWarning,
                                         check_stacklevel=False):
+
             with ignore_xlrd_time_clock_warning():
```

Comment:
rather than doing this which makes this *very* complicated to grok. I would have base class tests (most of them that are generic); then you can have specific tests for xlrd & openpyxl. I would restructure these tests in a precursor (just for xlrd)

Reply:
alternatively you can do things like

```
if engine == 'xlrd':
    ...
elif engine == 'openpyxl'
 ....
```

is prob ok; but using nested decorators which suppress things is a no-no